litex/soc/interconnect/csr.py

```diff
@@ -239,6 +309,12 @@ def __init__(self, size=1, reset=0, atomic_write=False, write_from_dev=False, al
             self.we = Signal()
             self.dat_w = Signal(self.size - self.alignment_bits)
             self.sync += If(self.we, self.storage_full.eq(self.dat_w << self.alignment_bits))
+        for field in [*fields]:
+            field_assign = getattr(self.fields, field.name).eq(self.storage[field.offset:field.offset + field.size])
+            if field.pulse:
+                self.comb += If(self.storage.re, field_assign)
```

Comment:
The `self.storage.re` signal should be `Replicate()`d out to `field.size`, though in reality I can't think of a usecase where `pulse` would be anything but a 1-bit signal.

Reply:
In fact the way we are doing it here, we do the full assignment on `self.storage.re`, so it should be fine, but yes, we should probably limit pulse usage to 1-bit fields.

Reply:
I see.  That's a much better way of doing it than I was originally doing it.